core/src/avm2.rs

```diff
@@ -579,6 +579,39 @@ impl<'gc> Avm2<'gc> {
         Ok(())
     }
 
+    pub fn lookup_class_for_symbol(
+        activation: &mut Activation<'_, 'gc>,
+        domain: Domain<'gc>,
+        name: QName<'gc>,
+    ) -> Result<ClassObject<'gc>, Error<'gc>> {
+        // Note: The panic can be hit if a script registers class A but also has
+        // a trait A that doesn't refer to the class. However, this is extremely
+        // difficult to craft, so just hope that we won't come across it.
+        let class_object = domain
+            .get_defined_value(activation, name)?
+            .as_object()
+            .and_then(|o| o.as_class_object())
+            .expect("Value referred to by SymbolClass should be ClassObject");
+
+        // Ensure this name actually refers to a class
+        let class = domain
+            .get_class(activation.context, &name.into())
+            .ok_or_else(|| {
+                make_error_1014(
+                    activation,
+                    Error1014Type::ReferenceError,
+                    name.to_qualified_name(activation.gc()),
+                )
+            })?;
+
+        // See first comment
+        if class_object.inner_class_definition() != class {
```

Comment:
To be sure, what does FP do here? Does it also attempt both and compares, or does it try just one of these?

Reply:
I have no idea; I couldn't test with FFDEC and I don't want to try hex-editing a SWF to figure it out- it's very difficult to trigger anyway, and if we do find a SWF that tries it, we'll get an error report

Reply:
I'll add the proper behavior when the ClassObject wasn't initialized properly (set to `null`) to avoid a panic in that case, since it's easy to cause